examples/platform/silabs/efr32/rs911x/rsi_if.c

```diff
@@ -485,15 +529,42 @@ static void wfx_rsi_do_join(void)
                 wfx_rsi.dev_state &= ~WFX_RSI_ST_STA_CONNECTING;
                 WFX_RSI_LOG("%s: rsi_wlan_connect_async failed with status: %02x on try %d", __func__, status,
                             wfx_rsi.join_retries);
-                vTaskDelay(4000);
-                /* TODO - Start a timer.. to retry */
+
+                if (!is_disconnection_event)
+                {
+                    /* After the reboot or a commissioning time device failed to connect with AP.
+                     * Device will retry to connect with AP upto WFX_RSI_CONFIG_MAX_JOIN retries.
+                     */
+                    WFX_RSI_LOG("%s: Next attempt after %d Seconds", __func__, CONVERT_MS_TO_SEC(WLAN_RETRY_TIMER_MS));
+                    vTaskDelay(pdMS_TO_TICKS(WLAN_RETRY_TIMER_MS));
+                }
+                else
+                {
+                    /* After disconnection
+                     * At the telescopic time interval device try to reconnect with AP, upto WLAN_MAX_RETRY_TIMER_MS
+                     * intervals are telescopic. If interval exceed WLAN_MAX_RETRY_TIMER_MS then it will try to
+                     * reconnect at WLAN_MAX_RETRY_TIMER_MS intervals.
+                     */
+                    if (retryInterval > WLAN_MAX_RETRY_TIMER_MS)
+                    {
+                        retryInterval = WLAN_MAX_RETRY_TIMER_MS;
+                        WFX_RSI_LOG("%s: Next attempt after %d Seconds", __func__, CONVERT_MS_TO_SEC(WLAN_MAX_RETRY_TIMER_MS));
+                    }
+                    else
+                    {
+                        WFX_RSI_LOG("%s: Next attempt after %d Seconds", __func__, CONVERT_MS_TO_SEC(retryInterval));
+                    }
+                    vTaskDelay(pdMS_TO_TICKS(retryInterval));
+                    retryInterval += retryInterval;
+                }
+                wfx_rsi.join_retries++;
```

Comment:
Pattern#2

Reply:
Pattern https://github.com/project-chip/connectedhomeip/pull/1 and https://github.com/project-chip/connectedhomeip/issues/2 are slight different whereas for pattern https://github.com/project-chip/connectedhomeip/pull/1 and https://github.com/project-chip/connectedhomeip/pull/3 are same but used in rs9116 and wf200 resp so only one file will get complied at a time.

In pattern https://github.com/project-chip/connectedhomeip/pull/1 we have an extra if condition to check retries before giving the delay whereas in pattern https://github.com/project-chip/connectedhomeip/issues/2 we are directly giving the delay without any condition.